src/ini.rs

```diff
@@ -779,41 +790,52 @@ impl Ini {
                     .or_insert_with(|| Some(String::new()));
 
                 match val {
-                    Some(x) => {
-                        x.push_str(LINE_ENDING);
-                        x.push_str(trimmed);
+                    Some(s) => {
+                        for _ in 0..blank_lines {
+                            s.push_str(LINE_ENDING);
+                        }
+                        s.push_str(LINE_ENDING);
+                        s.push_str(trimmed);
                     }
                     None => {
-                        *val = Some(format!("{}{}", LINE_ENDING, trimmed));
+                        let mut s = String::with_capacity(
+                            (blank_lines + 1) * LINE_ENDING.len() + trimmed.len(),
+                        );
+                        for _ in 0..blank_lines {
+                            s.push_str(LINE_ENDING);
+                        }
+                        s.push_str(LINE_ENDING);
+                        s.push_str(trimmed);
+                        *val = Some(s);
                     }
                 }
+            } else {
+                let valmap = map.entry(section.clone()).or_default();
 
-                continue;
-            }
```

Comment:
The diff looks more invasive than it is. Instead of using a `continue` in the `if`, I removed the `continue`, and put the else path in an `else`.

(The motivation here is that we now have `blank_lines = 0;` at the end of the loop; if we retain the `continue`, we need to duplicate that reset code and ensure we preserve it in any other early terminations.)